tools/src/main/java/org/apache/kafka/trogdor/coordinator/CoordinatorRestResource.java

```diff
@@ -96,13 +102,24 @@ public Empty destroyTask(@DefaultValue("") @QueryParam("taskId") String taskId)
     }
 
     @GET
-    @Path("/tasks")
-    public TasksResponse tasks(@QueryParam("taskId") List<String> taskId,
+    @Path("/tasks/")
+    public Response tasks(@QueryParam("taskId") List<String> taskId,
             @DefaultValue("0") @QueryParam("firstStartMs") long firstStartMs,
             @DefaultValue("0") @QueryParam("lastStartMs") long lastStartMs,
             @DefaultValue("0") @QueryParam("firstEndMs") long firstEndMs,
-            @DefaultValue("0") @QueryParam("lastEndMs") long lastEndMs) throws Throwable {
-        return coordinator().tasks(new TasksRequest(taskId, firstStartMs, lastStartMs, firstEndMs, lastEndMs));
+            @DefaultValue("0") @QueryParam("lastEndMs") long lastEndMs,
+            @DefaultValue("") @QueryParam("state") String state) throws Throwable {
```

Comment:
Can't you just make this `Optional<TaskStateType> state`?  Or does jetty not handle Optionals yet?

Reply:
It does not handle them as far as I could tell and google. I tried to but hit errors